src/qililab/qprogram/qblox_compiler.py

```diff
@@ -454,6 +454,7 @@ def _handle_play(self, element: Play):
             index_I, index_Q, duration = self._append_to_waveforms_of_bus(
                 bus=element.bus, waveform_I=waveform_I, waveform_Q=waveform_Q
             )
+            duration = element.wait_time or duration  # TODO: Change this in clean fix
```

Comment:
This `or` operator between objects can be sketchy, so I'm just gonna write this here to help we are choosing the good one we wanna prioritize in each case :) 

Def: `variable = x or y` gives `x`, if it evaluates to `True`, otherwise `y`. 

And in our case (`int | None`) it will evaluate to `False` when we have `None` or a `0`

So in this case, if `element.wait_time` is not `None` or a `0` we will keep it, without checking the previous value of `duration`, and if `element.wait_time` is `None` or a `0`, then we will keep whatever `duration` was previously, doesn't matter what it was, right?

Reply:
`A = B or C` is the same as writing

```
if B is not None:
   A = B
else:
   B = C
```

In our case, if the user has provided `element.wait_time` I want to use that as duration, otherwise keep duration as it is.

Reply:
You are right about the case `wait_time == 0`. I will change the syntax to normal.

Reply:
Thats not exactly what I read online, what I read, is more like: 
```
if B not in [None, 0, 0.0, {}, [], ...]:
   A = B
else:
   A = C
```

So for the 0 case, it wouldn't overwrite it, sorry I had edited the comment just when you were answering, since my point wasn't clear enough I think 😬

Reply:
Okei I just saw your second answer, cool! 🚀 

Sorry for the late edit 🙌